### DISCUSSION TITLE
Help: referencing parent field inside definition

### DISCUSSION THREAD

**Discussion by b4nst** [cuelang/cue#430] [General]
I want to reference a field from the parent scope of a definition, is there a way to do that ?

```cue
#Definition: {
  name: string
  nested: {
    name: name // Reference itself instead of the root name
  }
}
```

Obviously the problem here is that the nested field has the same name than the parent field, so it resolve to itself. Currently the only fix I found was to create an hidden alias field:

```cue
#Definition: {
  name: string
  _hidden: name

  nested: {
    name: _hidden
  }
}
```

It's working but it would be nice if I could directly reference the root field in some way.

**Comment by seh** [comment]
An alias would probably help here. If you try that, your experience might intersect with mine in #380, though there's no embedding here.

**Comment by b4nst** [comment]
Yeah, actually I just found #380 and saw your ref to my issue after finishing reading it 😂. Indeed, there's no embedding here, but the alias/self reference would be really helpful in that case too. I think I've tried tried every syntax guess (like `#Definition.name`)

**Comment by seh** [comment]
It's like learning C: "Did you try adding an asterisk? How many did you add? Try another one."

**Comment by myitcv** [comment]
FYI, we have [discussions](https://github.com/cuelang/cue/discussions) enabled for this repo, helping the GitHub folks to try out that beta. This is a perfect candidate for a discussion. Indeed, I'm going to try and press a button now that will convert this to a discussion. Fingers crossed it will work... apologies in advance if it doesn't!

**Reply by myitcv** [reply]
I _think_ that worked. Please let us know if anything _doesn't_ work as a result.

**Reply by b4nst** [reply]
Looks perfectly fine ! Thanks for pointing me to the `discussions`, I don't have yet the reflex to check that tab.

**Reply by myitcv** [reply]
> I don't have yet the reflex to check that tab.

That's incidentally one of the big "questions" - how do people know when to create a discussion vs issue. Hence why the ability to migrate one to the other is invaluable. Things will improve over time, such as the ability to label discussions in just the same way that we can label (and as a result categorise) issues.

**Reply by seh** [reply]
Yes, it's hard to know if you're failing because you don't know CUE well enough, or because you've discovered a bug in CUE. The former sounds like it needs a discussion. The latter sounds like it needs a defect report.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/430.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
